pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1210,6 +1211,11 @@ public PulsarClient getReplicationClient(String cluster, Optional<ClusterData> c
                     clientBuilder.listenerName(data.getListenerName());
                     log.info("Configuring listenerName {}", data.getListenerName());
                 }
+
+                // Set memory limit for replication client
+                clientBuilder.memoryLimit(pulsar.getConfiguration().getReplicationClientMemoryLimitBytes(),
+                        SizeUnit.BYTES);
```

Comment:
This is only one of several places where we configure the pulsar client in the broker.

Reply:
The other location in PulsarService is now handled too.